src/datatypes/avro.rs

```diff
@@ -0,0 +1,95 @@
+//! Avro data types related functions.
+
+use crate::datatypes::{Any, Field, Primitive, Schema};
+use crate::error::Result;
+use crate::{Error, ErrorKind};
+use apache_avro::schema::{Name, RecordField as AvroRecordField, RecordFieldOrder, RecordSchema};
+use apache_avro::Schema as AvroSchema;
+use std::collections::BTreeMap;
+
+impl<'a> TryFrom<&'a Schema> for AvroSchema {
```

Comment:
Please implement those logic in `types::to_avro` instead. We should not expose them to users directly.

icelake has concepts of `in_memory` and `on_disk` from day 1:

- Users SHOULD only operate on `in_memory` structs.
- Users CAN convert icelake `in_memory` structs at the API bounary
  - From/Into to other crates in_memory structs like `arrow`/`avro`
  - Parse/Format to other formats like table metadata files and manifest files.

All the internal convertion structs are hidden from users.